### ISSUE TITLE
Updates for Selection Model

### ISSUE THREAD

**Issue by bryevdv** [bokeh/bokeh#7611]
After #7261 there is still some follow on work to do:

* update all examples to use new `Selection` model
* expand docs around the `Selection` model
* Add a migration section to release notes (programmatically updated selections will break)

**Comment by clairetang6** [comment]
Duplicate of [issue 7516](https://github.com/bokeh/bokeh/issues/7516). I'll add that issue to the milestone and add checkboxes for the tasks.

**Comment by bryevdv** [comment]
Ah I missed that issue, thanks for the catch!
